File: storage/db-core/src/main/java/com/coffee/gu/TransactionHistoryRepositoryImpl.java
package com.coffee.gu;

import com.coffee.gu.transaction.TransactionHistory;
import com.coffee.gu.transaction.TransactionHistoryRepository;
import org.springframework.stereotype.Repository;

@Repository
public class TransactionHistoryRepositoryImpl implements TransactionHistoryRepository{

    private final TransactionHistoryJpaRepository transactionHistoryJpaRepository;

    public TransactionHistoryRepositoryImpl(TransactionHistoryJpaRepository transactionHistoryJpaRepository) {
        this.transactionHistoryJpaRepository = transactionHistoryJpaRepository;
    }

    @Override
    public TransactionHistory save(TransactionHistory transactionHistory) {
        return transactionHistoryJpaRepository.save(TransactionHistoryEntity.from(transactionHistory)).toModel();
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/transaction/TransactionHistoryRepositoryImpl.kt
package com.coffee.gu.transaction

import org.springframework.stereotype.Repository

@Repository
class TransactionHistoryRepositoryImpl(
    private val transactionHistoryJpaRepository: TransactionHistoryJpaRepository,
) : TransactionHistoryRepository {

    override fun save(transactionHistory: TransactionHistory): TransactionHistory {
        return transactionHistoryJpaRepository.save(TransactionHistoryEntity.from(transactionHistory)).toModel()
    }
}
